src/crimson/osd/pg_backend.cc

```diff
@@ -209,13 +209,19 @@ PGBackend::read(const ObjectState& os, OSDOp& osd_op)
   return _read(oi.soid, offset, length, op.flags).safe_then(
     [&oi, &osd_op](auto&& bl) -> read_errorator::future<> {
     if (!_read_verify_data(oi, bl)) {
-      return crimson::ct_error::object_corrupted::make();
+      // crc mismatches
+      return crimson::ct_error::eagain::make();
     }
     logger().debug("read: data length: {}", bl.length());
     osd_op.rval = bl.length();
     osd_op.outdata = std::move(bl);
     return read_errorator::now();
-  });
+  }).handle_error(
+      crimson::ct_error::input_output_error::handle([]() -> crimson::errorator<crimson::ct_error::eagain>::future<> {
```

Comment:
@AmnonHanuhov could you please reconsider https://github.com/ceph/ceph/commit/022525138759bb435a408a57f0629e821e786e57 ?

Reply:
We reverted back to object_corrupted so using eagain here is not valid

Reply:
@AmnonHanuhov if you take a closer look, there are two other changes in my commit

1. drop the return type . see https://github.com/ceph/ceph/commit/022525138759bb435a408a57f0629e821e786e57#diff-57f32ce881e4f585be2b2b2d3c5ef543db9d68689f8c118e7ff71b1741861314R219
2. do not use `handle_error(`, for more compact code